src/matrix/matrixGetSubMatrix.ts

```diff
@@ -0,0 +1,34 @@
+import { DoubleMatrix } from '../types';
+import { matrixCreateEmpty } from './index';
+import { matrixCheckRanges } from './matrixCheckRanges';
+
+/**
+ * Get a subMatrix from matrix, the function check if the subMatrix
+ * lies into the dimensions of matrix.
+ * @param matrix - matrix that will receive the new element values.
+ * @param startRow - row index in matrix for the first row in subMatrix.
+ * @param startColumn - column index in matrix for the first column in subMatrix.
+ * @returns The modified `matrix`.
+ */
+export function matrixGetSubMatrix(
+  matrix: DoubleMatrix,
+  options: {
```

Comment:
You need to create a specific type 'MatrixGetSubMatrixOptions' that describe each property. You current jsdoc is wrong as well (you forgot 2 properties)

I would also add default values for all the properties. 0 for the from and length -1 for the end.

Please also be clear if you include the last element or not.